### PR TITLE
Added Additional column in audit discovery report

### PR DESCRIPTION
**Pull request by ankit-privado**
Added `File name`, `Member Type`, `Collection endpoint path`, `Collection method full name` in element discovery sheet. 

Ticket: https://privado-ai.atlassian.net/browse/LE-321

### CODE REVIEW

File: src/main/scala/ai/privado/audit/DataElementDiscovery.scala [hunk 1]
```diff
@@ -151,33 +186,71 @@ object DataElementDiscovery {
     }
 
     // Header List
-    workbookResult += List("Class", "Member", "Tagged", "Source Rule ID", "Input to Collection")
+    workbookResult += List(
+      "Class",
```
Comment (pandurangpatil) [inline src/main/scala/ai/privado/audit/DataElementDiscovery.scala:190]: create constants for these literals. Check with @khemrajrathore

File: src/main/scala/ai/privado/audit/DataElementDiscovery.scala [hunk 2]
```diff
@@ -151,33 +186,71 @@ object DataElementDiscovery {
...
+          val isCollectionInput = if (collectionInputList.contains(key.fullName)) "YES" else "NO"
+          val collectionMethodName =
+            if (collectionMethodInfo.contains(key.fullName)) collectionMethodInfo(key.fullName).methodName else "--"
+          val collectionEndPointPath =
+            if (collectionMethodInfo.contains(key.fullName)) collectionMethodInfo(key.fullName).endpoint else "--"
+          if (taggedMemberInfo.contains(key.fullName)) {
+            workbookResult += List(
+              key.fullName,
+              key.file.head.name,
+              "--",
```
Comment (pandurangpatil) [inline src/main/scala/ai/privado/audit/DataElementDiscovery.scala:214]: Same here create constants

Comment (ankit-privado) [reply src/main/scala/ai/privado/audit/DataElementDiscovery.scala]: done

**Review by pandurangpatil — DISMISSED** [review]
looks good to me

File: src/main/scala/ai/privado/audit/DataElementDiscovery.scala [hunk 3]
```diff
@@ -83,16 +82,19 @@ object DataElementDiscovery {
   }
 
   // Get list of member variable present in given class
-  def getMemberUsingClassName(xtocpg: Try[Cpg], classNameSet: Set[String]): mutable.Map[String, List[Member]] = {
+  def getMemberUsingClassName(xtocpg: Try[Cpg], classNameSet: Set[String]): mutable.Map[TypeDecl, List[Member]] = {
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/audit/DataElementDiscovery.scala:85]: Return immutableMap instead

File: src/main/scala/ai/privado/audit/DataElementDiscovery.scala [hunk 4]
```diff
@@ -83,16 +82,19 @@ object DataElementDiscovery {
...
     xtocpg match {
       case Success(cpg) => {
         classNameSet.foreach(className => {
           // Get member variable of class and put it into map
-          val memberInfo = cpg.typeDecl.filter(_.order > 0).where(_.fullName(className)).member.toList
-          memberInfoMap.put(className, memberInfo)
+          val classInfo = cpg.typeDecl.filter(_.order > 0).where(_.fullName(className)).l
+          if (classInfo.nonEmpty) {
+            val memberInfo = classInfo.head.member.toList
+            memberInfoMap.put(classInfo.head, memberInfo)
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/audit/DataElementDiscovery.scala:96]: You can write these entire Success lines as 
```
classNameSet.map(className => { cpg.typeDecl.where(_.fullName(className)).map(typeDeclNode => (typeDeclNode, typeDeclNode.member.l)).toMap})
```

File: src/main/scala/ai/privado/audit/DataElementDiscovery.scala [hunk 5]
```diff
@@ -126,11 +128,41 @@ object DataElementDiscovery {
     collectionInputList.toList
   }
 
+  def getCollectionMethodInfo(xtocpg: Try[Cpg]): Map[String, ListBuffer[CollectionMethodInfo]] = {
+    // className -> (MethodName, Endpoint)
+    val collectionMethodInfoMap = new mutable.HashMap[String, ListBuffer[CollectionMethodInfo]]()
+    xtocpg match {
+      case Success(cpg) => {
+        val parameterList = cpg.parameter.where(_.tag).l
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/audit/DataElementDiscovery.scala:136]: For collectionPoints parameter, update the query to 
```
cpg.parameter.where(_.method.tag.nameExact(CatLevelOne.Collections.name)).l
```

Update this query every where

**Review by pandurangpatil — APPROVED** [review]
look good to me

**Pull request merged**